File: 2주차/SWEA1868. 파핑파핑 지뢰찾기/백동승.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.BufferedWriter;
import java.io.OutputStreamWriter;
  
class Solution {
  
    private static char[][] board = new char[302][302];
  
    private static void dfs(int r, int c) {
        if (board[r][c] == '.') {
            board[r][c] = 'C';
            for (int off_r = -1; off_r <= 1; off_r++)
                for (int off_c = -1; off_c <= 1; off_c++)
                    dfs(r + off_r, c + off_c);
        } else if (board[r][c] == '1') {
            board[r][c] = 'C';
        }
    }
  
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
  
        int T = Integer.parseInt(br.readLine());
  
        for (int test_case = 1; test_case <= T; test_case++) {
            int n;
            int click = 0;
            n = Integer.parseInt(br.readLine());
  
            for (int j = 0; j <= n + 1; j++)
                board[0][j] = '*';
            for (int i = 1; i <= n; i++) {
                board[i][0] = '*';
                String s = br.readLine();
                for (int j = 1; j <= n; j++)
                    board[i][j] = s.charAt(j-1);
                board[i][n + 1] = '*';
            }
            for (int j = 0; j <= n + 1; j++)
                board[n + 1][j] = '*';
  
            for (int i = 1; i <= n; i++)
                for (int j = 1; j <= n; j++)
                    if (board[i][j] == '*')
                        for (int off_i = -1; off_i <= 1; off_i++)
                            for (int off_j = -1; off_j <= 1; off_j++)
                                if (board[i + off_i][j + off_j] != '*')
                                    board[i + off_i][j + off_j] = '1';
              
            for (int i = 1; i <= n; i++)
                for (int j = 1; j <= n; j++)
                    if (board[i][j] == '.') {
                        dfs(i, j);
                        click++;
                    }
  
            for (int i = 1; i <= n; i++)
                for (int j = 1; j <= n; j++)
                    if (board[i][j] == '1')
                        click++;
  
            bw.write("#" + test_case + " " + click + "\n");
        }
        bw.flush();
        bw.close();
    }
}

File: 2주차/1868. 파핑파핑 지뢰찾기/백동승.py
def dfs(r, c):
    if board[r][c] == '.' :
        board[r][c] = 'C';
        for off_r in range(-1,2):
            for off_c in range(-1,2):
                dfs(r+off_r, c+off_c)
    elif board[r][c] == '1':
        board[r][c] = 'C';
 
T = int(input())
                 
for test_case in range(1, T + 1):
    n = int(input())
    board = [['*' for i in range(n + 2)]]
    for i in range(n):
        board.append(list('*' + input() + '*'))
    board.append(['*' for i in range(n + 2)])
     
    for i in range(1,n+1):
        for j in range(1,n+1):
            if board[i][j] == '*':
                for off_i in range(-1,2):
                    for off_j in range(-1,2):
                        if board[i+off_i][j+off_j] != '*':
                            board[i+off_i][j+off_j] = '1'
     
    click = 0
    for i in range(1,n+1):
        for j in range(1,n+1):
            if board[i][j] == '.' :
                dfs(i,j)
                click += 1
             
    for i in range(1,n+1):
        for j in range(1,n+1):
            if board[i][j] == '1' :
                click += 1
     
    print("#{} {}".format(test_case, click))
